src/Executables/PrintMinimumGridSpacing/PrintMinimumGridSpacing.hpp

```diff
@@ -0,0 +1,258 @@
+      tmpl::list<Parallel::PhaseActions<typename Metavariables::Phase,
+                                        Metavariables::Phase::Initialization,
+                                        tmpl::list<>>>;
+
+  static void initialize(
+      Parallel::CProxy_ConstGlobalCache<Metavariables>& /*global_cache*/) {}
+
+  static void execute_next_phase(
+      const typename Metavariables::Phase /*next_phase*/,
+      const Parallel::CProxy_ConstGlobalCache<
+          Metavariables>& /*global_cache*/) {}
+};
+
+template <size_t Dim, typename Metavariables>
+struct ElementArray {
+  using chare_type = Parallel::Algorithms::Array;
+  using metavariables = Metavariables;
+  using elemental_min_grid_spacing = double;
+  using array_index = ElementIndex<Dim>;
+  using const_global_cache_tag_list = tmpl::list<>;
+  using options = tmpl::list<OptionTags::DomainCreator<Dim, Frame::Inertial>>;
+  using phase_dependent_action_list = tmpl::list<
+      Parallel::PhaseActions<typename Metavariables::Phase,
+                             Metavariables::Phase::Initialization,
+                             tmpl::list<Actions::InitializeElement<Dim>>>>;
+
+  using add_options_to_databox =
+      typename Actions::InitializeElement<Dim>::AddOptionsToDataBox;
+
+  static void initialize(
+      Parallel::CProxy_ConstGlobalCache<Metavariables>& global_cache,
+      const std::unique_ptr<DomainCreator<Dim, Frame::Inertial>>
+          domain_creator) noexcept {
+    auto& local_cache = *(global_cache.ckLocalBranch());
+    auto& element_array =
+        Parallel::get_parallel_component<ElementArray>(local_cache);
+
+    auto domain = domain_creator->create_domain();
+    for (const auto& block : domain.blocks()) {
+      const auto initial_ref_levs =
+          domain_creator->initial_refinement_levels()[block.id()];
+      const std::vector<ElementId<Dim>> element_ids =
+          initial_element_ids(block.id(), initial_ref_levs);
+      int which_proc = 0;
+      const int number_of_procs = Parallel::number_of_procs();
+      for (size_t i = 0; i < element_ids.size(); ++i) {
+        element_array(ElementIndex<Dim>(element_ids[i]))
+            .insert(global_cache,
+                    {domain_creator->initial_extents(),
+                     domain_creator->create_domain()},
+                    which_proc);
+        which_proc = which_proc + 1 == number_of_procs ? 0 : which_proc + 1;
+      }
+    }
+    element_array.doneInserting();
+  }
+
+  static void execute_next_phase(
+      const typename Metavariables::Phase next_phase,
+      Parallel::CProxy_ConstGlobalCache<Metavariables>& global_cache) {
+    auto& local_cache = *(global_cache.ckLocalBranch());
+    auto& element_array =
+        Parallel::get_parallel_component<ElementArray>(local_cache);
+    if (next_phase == Metavariables::Phase::CallArrayReduce) {
+      Parallel::simple_action<Actions::ArrayReduce<Dim>>(element_array);
+    }
+    else {
+        element_array.start_phase(next_phase);
+    }
```

Comment:
To make the control flow through this executable more clear, I'd prefer to make the action that contributes to the reduction from each element an "iterable action" instead of a "simple action". That means you don't invoke it here in `execute_next_phase` (just always call `start_phase`), but instead you make another `PhaseActions` in the `phase_dependent_action_list` above with this action. I find this makes it way easier to understand what a parallel component is doing, because it's apparent from reading the `phase_dependent_action_list`.

To make your action an "iterable action", adjust the `apply` function signature (see e.g. your `InitializeElement::apply` function) and return a `std::tuple<DataBox, bool>`. You can return `true` for the `bool` to terminate the phase directly with this action, or return `false` and add `Parallel::Actions::TerminatePhase` to the phase actions (I think I prefer the former in this simple case).